From fc20a3670c897f23362916dda7a5a5bc40f1f1c7 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Wed, 5 Feb 2020 12:01:48 +0800
Subject: [PATCH 1/2] drm/bridge: adv7511: destroy changeset after changeset
 applied

The initialized changeset should be destroyed after changeset applied.
And of_changeset_update_property() will malloc memory to
of_changeset_entry, it should be freed also after changeset applied.

To fix below call trace:
[<0000000016762abf>] kmem_cache_alloc_trace+0x2a8/0x2f8
[<0000000075e85f52>] of_changeset_action+0x44/0xb8
[<000000008e88a239>] adv7511_probe+0x698/0x9d0
[<000000005d316d77>] i2c_device_probe+0x218/0x278
[<0000000051186a2b>] driver_probe_device+0x210/0x2f8
[<00000000b3560bd1>] __driver_attach+0xf4/0xf8
[<000000002dd178d4>] bus_for_each_dev+0x84/0xd8
[<00000000c47daad4>] driver_attach+0x30/0x40
[<00000000e016b28d>] bus_add_driver+0x1e0/0x210
[<00000000a015e1f2>] driver_register+0x64/0x110
[<00000000b3cfa40d>] i2c_register_driver+0x4c/0x90
[<00000000586afb19>] adv7511_init+0x34/0x40
[<0000000062bbd9eb>] do_one_initcall+0x68/0x248
[<000000008e7477db>] kernel_init_freeable+0x2d4/0x3cc
[<00000000e1151112>] kernel_init+0x18/0x108
[<0000000047c04d0e>] ret_from_fork+0x10/0x1c

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/gpu/drm/bridge/adv7511/adv7511_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
 mode change 100755 => 100644 drivers/gpu/drm/bridge/adv7511/adv7511_drv.c

diff --git a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
old mode 100755
new mode 100644
index 67ebbfe0be9d..ac31a0f82a79
--- a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
+++ b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
@@ -1354,7 +1354,7 @@ static int adv7511_probe(struct i2c_client *i2c, const struct i2c_device_id *id)
 			dev_warn(dev,
 				"Probe failed. Remote port '%s' disabled\n",
 				remote_node->full_name);
-
+		of_changeset_destroy(&ocs);
 		of_node_put(remote_node);
 	};
 #endif
-- 
2.17.1

